packages/runtime/container-runtime/src/test/gc/garbageCollection.spec.ts

```diff
@@ -222,13 +224,109 @@ describe("Garbage Collection Tests", () => {
 			() => {
 				closeCalled = true;
 			},
-		) as GcWithPrivates;
+		);
 		assert(
 			closeCalledAfterExactTicks(defaultSessionExpiryDurationMs),
 			"Close should have been called at exactly defaultSessionExpiryDurationMs",
 		);
 	});
 
+	describe("runSweepPhase", () => {
+		it("Tombstone then Delete", async () => {
+			// Simple starting reference graph - root and two nodes
+			defaultGCData.gcNodes["/"] = [nodes[0], nodes[1]];
+			defaultGCData.gcNodes[nodes[0]] = [];
+			defaultGCData.gcNodes[nodes[1]] = [];
+
+			// Sweep enabled
+			gc = createGarbageCollector({ gcOptions: { [gcSweepGenerationOptionName]: 1 } });
+			const spies = {
```

Comment:
What does a spy do? 
Can you please add a comment here what this is supposed to do for clarity?

Reply:
It's a way to spy on these functions.  Think of it like - if these functions were injected dependencies we could easily build a fake one that jots down how it's called.  This is like that, but without having to structure the actual code with DI, and also the original functions still run.  There are other things called Stubs and Mocks that have slightly different behavior, I honestly don't understand it all.  But this was what I was looking for so I was glad to find it.